src/main/java/it/gov/pagopa/paymentupdater/service/PaymentService.java

```diff
@@ -1,21 +1,24 @@
 package it.gov.pagopa.paymentupdater.service;
 
-import java.util.Map;
+import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.ExecutionException;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 
+import it.gov.pagopa.paymentupdater.dto.ProxyResponse;
 import it.gov.pagopa.paymentupdater.model.Payment;
 
 public interface PaymentService {
 
-	Optional<Payment> getPaymentByNoticeNumberAndFiscalCode(String noticeNumber, String fiscalCode);
-
 	void save(Payment reminder);
 
-	Map<String, String> checkPayment(Payment payment) throws JsonProcessingException, InterruptedException, ExecutionException;
+	ProxyResponse checkPayment(Payment payment) throws JsonProcessingException, InterruptedException, ExecutionException;
 
 	Optional<Payment> findById(String messageId);
+	
+	List<Payment> getPaymentsByRptid(String rptid);
+
+	int countFindById(String id);
```

Comment:
ditto

Reply:
As mentioned, if a duplicate message arrives, we check the existence of the payment. Could this check affect performance?

Reply:
fixed